client/client.go

```diff
@@ -393,3 +393,14 @@ func (client *Client) CreateUser(request *CreateUserRequest) (*CreateUserResult,
 
 	return &createResult, nil
 }
+
+// UpdateRevisions asks the daemon to sync the core/kernel snap revisions
+// with what was booted
+func (client *Client) UpdateRevisions() error {
+	var rsp interface{}
+	if _, err := client.doSync("POST", "/v2/update-revisions", nil, nil, nil, &rsp); err != nil {
```

Comment:
I initially did this when snapd started. However this turned out to be problematic, because the bootloader update happens only after the "multi-user" target was reached, i.e. only after all of the system has started up successfully. At this point the bootloader variables are updated and meaningful. One way is what I'm doing here, just triggering a update via the new endpoint. I agree that its awkward. Essentially we either need to prod the daemon at the right time somehow or we could start a loop inside the daemon that looks for the bootloader variable going from snapd_mode=trying to snapd_mode="". Once that transition happend the revisions can be updated. If you prefer the later approach, I'm happy to update the branch. Alternatively we could also move the bootloader update into the daemon too, so no "boot-ok" after multi-user, once we have a working daemon we consider the boot ok. Please let me know your thoughts on this :)